app/controller.go

```diff
@@ -456,22 +450,9 @@ func handleBlockRewards(validators *identity.ValidatorStore, rewards *rewards.Re
 	})
 
 	//Loop through all validators that participated in signing the last block
-	for index := 0; index < numOfSignatures; index++ {
-
-		//Get the validator's vote by index
-		validatorVote := blockCommit.GetVote(index)
-		if validatorVote == nil {
-			continue
-		}
-		if len(validatorVote.ValidatorAddress) == 0 {
-			continue
-		}
-		if len(validatorVote.Signature) == 0 {
-			continue
-		}
-
+	for _, vote := range votes {
 		//Verify Validator Address
-		valAddress := keys.Address(validatorVote.ValidatorAddress)
+		valAddress := keys.Address(vote.Validator.Address)
 		if valAddress.Err() != nil {
```

Comment:
We get the Validator address from tendermint , wondering if this error can ever happen ?